client/actions/events.js

```diff
@@ -690,22 +690,68 @@ const previewEvent = (event) => ({
     payload: get(event, '_id'),
 })
 
+const _unlockAndOpenEventDetails = (event) => (
+    (dispatch, getState, { api, notify }) => (
+        api('events_unlock', event).save({})
+        .then(() => {
+            // Call openPlanningEditor to obtain a new lock for editing
+            dispatch(_openEventDetails(event))
+        }, (error) => {
+            const msg = get(error, 'data._message') || 'Could not unlock on the event.'
+            notify.error(msg)
+            return Promise.reject()
+        })
+    )
+)
+
 /**
  * Opens the Edit Event panel with the supplied Event
  * @param {object} event - The Event ID to edit
  * @return Promise
  */
-const _openEventDetails = (event) => ({
-    type: EVENTS.ACTIONS.OPEN_EVENT_DETAILS,
-    payload: get(event, '_id', event || true),
-})
+const _openEventDetails = (event) => (
+    (dispatch, getState, { api, notify }) => {
+        const id = get(event, '_id')
+        if (id) {
+            api.save('events_lock', {}, { lock_action: 'edit' }, { _id: id }).then((item) => {
+                dispatch({
+                    type: EVENTS.ACTIONS.OPEN_EVENT_DETAILS,
+                    payload: item,
+                })
+            }, (error) => {
+                const msg = get(error, 'data._message') || 'Could not lock the planning item.'
+                notify.error(msg)
+                dispatch({
+                    type: EVENTS.ACTIONS.OPEN_EVENT_DETAILS,
+                    payload: id,
+                })
+            })
+
+        } else {
+            dispatch({
+                type: EVENTS.ACTIONS.OPEN_EVENT_DETAILS,
+                payload: true,
+            })
+        }
+    }
+)
 
 /**
  * Action to close the Edit Event panel
  * @return object
  */
-const closeEventDetails = () => (
-    { type: EVENTS.ACTIONS.CLOSE_EVENT_DETAILS }
+const closeEventDetails = (event) => (
+    (dispatch, getState, { api, notify }) => {
+        if (selectors.isEventDetailLockedInThisSession(getState())) {
+            api('events_unlock', event).save({})
```

Comment:
Personally I don't mind if there are actions that don't update the state now, because they could one day.
My point was to avoid to see the same API and `notify` calls in different functions.